python/rmm/_lib/torch_allocator.pyx

```diff
@@ -0,0 +1,19 @@
+from cuda.ccudart cimport cudaStream_t
+from libc.stdint cimport uintptr_t
+from libc.stdio cimport printf
+
+from rmm._lib.memory_resource cimport device_memory_resource
+from rmm._lib.per_device_resource cimport get_current_device_resource
+
+
+cdef public void* allocate(
+    ssize_t size, int device, void* stream
```

Comment:
Q: `device` is ignored by design? (I was reviewing https://github.com/cupy/cupy/pull/7210 and noticed this.)

Reply:
Ah, great catch! This brings out a subtle problem:

In RMM, each device has its own memory resource. Thus, to do the allocation on a specified `device` with RMM, I would write the torch `allocate` function like this:

cdef public void* allocate(ssize_t size, int device, void* stream) except * with gil:                                                                                                                                                                                       
    cdef device_memory_resource* mr = get_per_device_resource(device)                                                                                                                            
    return mr[0].allocate(size, <cudaStream_t> stream)

Unforunately, the deallocation function does not accept a `device` argument, so we cannot retrieve the memory resource that was used for allocation:

```c++
void deallocate(void* ptr, ssize_t size, void* stream)
```

I don't really see a way around this other than for the `deallocate` signature to include the `device` argument. cc: @emcastillo

Reply:
I would love to know too. TBH I am puzzled by PyTorch's (long-time) behavior of asking for `device`. It should just honor the current device...

Reply:
+1 to that

Reply:
I'll submit a follow-up PR adding support for `device`, once https://github.com/pytorch/pytorch/pull/91398 is merged.